core/blockchain.go

```diff
@@ -2555,6 +2540,35 @@ func (bc *BlockChain) UpdateStakingMetaData(tx *staking.StakingTransaction) erro
 	return nil
 }
 
+func (bc *BlockChain) addDelegationIndex(delegatorAddress, validatorAddress common.Address, root common.Hash) error {
+	delegations, err := bc.ReadDelegationsByDelegator(delegatorAddress)
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, delegation := range delegations {
+		if bytes.Compare(delegation.ValidatorAddress.Bytes(), validatorAddress.Bytes()) == 0 {
```

Comment:
suggest pull `validatorAddress.Bytes()` out of the loop to not byte thrash, copy memory over, over

Reply:
Done.